go/private/common.bzl

```diff
@@ -123,7 +123,7 @@ def go_importpath(ctx):
     path = path[1:]
   return path
 
-def env_execute(ctx, arguments, environment = None, **kwargs):
+def env_execute(ctx, arguments, environment = {}, **kwargs):
```

Comment:
Careful using mutable objects as default values: http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments

(I'm assuming this applies to .bzl the same as .py.)

Reply:
It does apply, but because of "frozen" mutable objects it's much less dangerous (as in you tend to end up mutating a frozen object and then get a runtime error rather than silent bad behavior). In general I avoid functions that modify their arguments anyway, it's generally not a great idea without very explicit signaling, and only then when copying and returning is expensive enough to be worth the special case. This function for instance always makes full copy before modifying, so it's safe.
That is also btw a truly hideous piece of language design...